lib/takeaway.rb

```diff
@@ -23,12 +26,12 @@ def see_menu
   end
 
   def choose_dish(name)
-    unless @menu[name].nil?
-      puts "You've chosen #{name}!"
-      @my_choices << name
-    else
+    if @menu[name].nil?
       puts "That's not a menu option! We don't sell that"
       @my_choices
+    else
```

Comment:
So, in the case of using an array of hashes you can 'push' the menu options as a number. But, again, just a different approach to the same result.